PacketDotNet/GrePacket.cs

```diff
@@ -105,5 +121,14 @@ public override string ToString(StringOutputType outputFormat)
 
             return buffer.ToString();
         }
+
+        /// <summary>
+        /// Custom parsers for GRE non-IANA assigned protocol numbers or proprietary protocols, e.g. ERSPAN.
+        /// The Key of dictionary is the protocol number. The value is the function of the custom parser.
+        /// The function must take two parameters and return the decoded payload.
+        /// The first parameter is the payload of the GRE packet. The second parameter is the GRE packet itself.
+        /// Returned value is the decoded payload as PacketOrByteArraySegment.
+        /// </summary>
+        public static Dictionary<ushort, Func<ByteArraySegment, GrePacket, PacketOrByteArraySegment>> CustomProtocolParser = new();
```

Comment:
I would prefer a CustomPayloadDecoder here, similar to this, for consistency:
https://github.com/dotpcap/packetnet/blob/af442b608f72d0c3f19ed4a9dc32f950558fde48/PacketDotNet/TransportPacket.cs#L30

You'd mostly have the same parameters (ByteArraySegment, GrePacket, PacketOrByteArraySegment).
The abstraction for protocol isn't needed, that can be done with the parameters passed in.

Reply:
Changed this part for consistency.